jmclient/jmclient/yieldgenerator.py

```diff
@@ -144,7 +143,7 @@ def on_tx_unconfirmed(self, offer, txid, removed_utxos):
         # announce new order, replacing the old order
         return [], [newoffers[0]]
 
-    def on_tx_confirmed(self, offer, confirmations, txid):
+    def on_tx_confirmed(self, offer, txid):
```

Comment:
This should have the same required arguments as the parent class `Maker`.

Reply:
OK so TIL that `@abc.abstractmethod` doesn't check anything other than a name! So that's a small gotcha for me at least.

Indeed both the function signatures for the `on_tx_[un]confirmed` variants had gone out of whack. It's additionally confusing that there are asymmetries between Taker and Maker here that I would forgive anyone for finding obtuse, but in any case I have fixed them up to actually match the base class signature as of https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/359/commits/820fc11cad9008a2ad50065bb3d0bb495b9809d2 (note the P2EP maker uses the function slightly differently).